### PR TITLE
Doc and refactor

### LINKED ISSUE #16: [Feature 6] Selection Documentation
Add more docstrings to classes and methods in the selection system. Convert some comments to docstrings.

### PR DESCRIPTION
**Pull request by ChowdhuryShaswata**
Added documentation for all public methods and classes related to the directory, selection or join systems.

Fixed a bug with the join controller that prevented opening in the event of no files being sent to the controller when sending it an intent.

### CODE REVIEW

File: app/src/main/java/com/example/notes2text/adapters/FileListAdaptor.java [hunk 1]
```diff
@@ -3,51 +3,78 @@
...
+     * @return FileViewHolder, the file holder.
+     */
+    @NonNull
     @Override
-    public ViewHolder onCreateViewHolder(ViewGroup source, int ViewType){
+    public FileViewHolder onCreateViewHolder(@NonNull ViewGroup source, int ViewType){
 
         View view = LayoutInflater.from(context).inflate(R.layout.file_holder_view_model, source, false);
-        return new ViewHolder(view);
+//        return new FileViewHolder(view);
```
Comment (JerryLi-617) [inline app/src/main/java/com/example/notes2text/adapters/FileListAdaptor.java]: Make sure to remove this line if its no longer needed.

**Review by JerryLi-617 — APPROVED** [review]
Good documentation, code is clear and easy to understand

**Review by CSEndeavour — APPROVED** [review]
Documentation does a good explaining how the code works.

File: app/src/main/java/com/example/notes2text/adapters/FileListAdaptor.java (hunk 1 above)
Comment (ChowdhuryShaswata) [reply app/src/main/java/com/example/notes2text/adapters/FileListAdaptor.java]: Fixed with commit.

**Review by julianmarrero — APPROVED** [review]
Documentation looks good, easy to understand going through the code and the functionality of methods

**Pull request merged**